docs/rhoas/rhoas-helm-guide/README.adoc

```diff
@@ -0,0 +1,395 @@
+
+.Procedure
+
+. Confirm that the Managed Service instances are set in the current context and are running.
+
++
+[source,shell]
+----
+$ rhoas context status
+----
+
+. Generate a yaml file that contains the connection configuration as a Kubernetes ConfigMap object.
++
+[source,shell]
+----
+$ rhoas generate-config --type configmap --output-file ./rhoas-services.yaml
+----
++
+. Apply the generated ConfigMap file.
++
+[source,shell]
+----
+$ oc apply -f “./rhoas-services.yaml”
+----
++
+. Consuming ConfigMap in a Kubernetes deployment.
++
+An application deployed as a Kubernetes deployment can have the ConfigMap values injected as environment variables.
++
+[source,shell]
+----
+env:
+  - name: KAFKA_HOST
+    valueFrom: 
+      configMapKeyRef:
+        name: my-context-3-configuration
+        key: kafka_host
+  - name: SERVICE_REGISTRY_URL
+    valueFrom: 
+      configMapKeyRef:
+        name: my-context-3-configuration
+        key: service_registry_url
+----
+. Consuming ConfigMap in a Helm Chart.
++
+Helm charts support templating which implies ConfigMap values can be passed into templates. This will allow developers to quickly deploy their applications to OpenShift clusters. All one will need is - the Helm chart, Kubernetes ConfigMap and Kubernetes Secret. The helm chart needs to follow the naming standards set by RHOAS CLI. 
++
+[source,shell]
+----
+env:
+  - name: KAFKA_HOST
+    valueFrom:
+      configMapKeyRef:
+        name: {{ .Values.rhoas.config }}
+        key: kafka_host
+	- name: SERVICE_REGISTRY_URL
+    valueFrom: 
+    configMapKeyRef:
+      name: {{ .Values.rhoas.config }}
+      key: service_registry_url
+----
++
+Helm renders the value for ConfigMap name and generates the manifest file for the deployment.
+The values can be fetched from the Values.yaml file which is native to helm, or a different file can be used by passing the file name to “--values” flag:
++
+[source,shell]
+----
+$ helm install . --generate-name --values my-values.yaml
+----
++
+The values can be set without using any file, by using the “--set-string” flag:
++
+[source,shell]
+----
+$ helm install . --generate-name --set-string rhoas.config=my-context-3-configuration
+----
+
+[id="proc-generating-credentials-for-application-services_{context}"]
+== Generate Credentials for Application Service Instances
+
+Applications make use of service-accounts to authenticate with the Kafka and Service Registry instances. RHOAS CLI can create a service-account and save the credentials as a Kubernetes Secret object.
```

Comment:
Application Services uses service accounts for authentication with Kafka and Service Registry instances. The RHOAS CLI can create a service account and save the credentials in a Kubernetes secret.